### PR TITLE
readersearch: do not show regexp checkbutton in pdf

### PR DESCRIPTION
**Pull request by hius07**
Minor optimization.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8450)

### CODE REVIEW

File: frontend/apps/reader/modules/readersearch.lua [hunk 1]
```diff
@@ -159,7 +160,9 @@ function ReaderSearch:onShowFulltextSearchInput()
             })
         end,
     }
-    self.input_dialog:addWidget(self.check_button_regex)
+    if not self.ui.document.info.has_pages then
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readersearch.lua:163]: Wouldn't the other way around make more sense?

Comment (hius07) [reply frontend/apps/reader/modules/readersearch.lua:163]: Don't understand the question, sorry.

Comment (Frenzie) [reply frontend/apps/reader/modules/readersearch.lua:163]: We don't want to say if not this and not that and not that when it's only supported in a single document format.

Comment (Frenzie) [reply frontend/apps/reader/modules/readersearch.lua:163]: Or you could add it to Document and use if hasFeature().

Comment (poire-z) [reply frontend/apps/reader/modules/readersearch.lua:163]: `if not self.ui.document.info.has_pages then` is our generic (ugly, but we have it everywhere) check for credocuments.
So, this feels just alright if you want this only for credocument :)

Comment (hius07) [reply frontend/apps/reader/modules/readersearch.lua:163]: Ah, I see. There are hundres of this check.
I assume it has been standardized long ago.
https://github.com/koreader/koreader/pull/7883#discussion_r658030644

Comment (Frenzie) [reply frontend/apps/reader/modules/readersearch.lua:163]: @hius07 There most certainly are not hundreds. Some do however exist.

```
$ rg "if not self.ui.document.info.has_pages"
frontend/apps/reader/modules/readerlink.lua
229:    if not self.ui.document.info.has_pages then

frontend/apps/reader/modules/readerdogear.lua
57:    if not self.ui.document.info.has_pages then

frontend/apps/reader/modules/readerbookmark.lua
1065:    if not self.ui.document.info.has_pages then

frontend/apps/reader/modules/readerhighlight.lua
137:    if not self.ui.document.info.has_pages then
679:    if not self.ui.document.info.has_pages then
920:    if not self.ui.document.info.has_pages and self.selected_text_start_xpointer then

frontend/apps/reader/modules/readerfooter.lua
2118:    if not self.ui.document.info.has_pages then
```

What you link is the reverse, which makes it very different. You should use if self.ui.rolling instead.

Comment (hius07) [reply frontend/apps/reader/modules/readersearch.lua:163]: > You should use if self.ui.rolling instead.

Is it applicable to this PR only?
Or to all of the above `not self.ui.document.info.has_pages` and possible future cases?

Comment (Frenzie) [reply frontend/apps/reader/modules/readersearch.lua:163]: My reasoning would likely apply to all cases (i.e., there is no guarantee that a future "rolling2" would not has_pages, MuPDF opening HTML, EPUB and FB2 already proving the point).

Comment (poire-z) [reply frontend/apps/reader/modules/readersearch.lua:163]: Let's update them from now on as we meet these `if (not) self.ui.document.info.has_pages` , starting with this PR.
```
if self.ui.rolling then
if self.ui.paging then
```
(and it's even more readable)

**Pull request merged**